core/trie2/triedb/database.go

```diff
@@ -21,30 +21,34 @@ type Database struct {
 	config *Config
 }
 
-func New(disk db.KeyValueStore, config *Config) *Database {
+func New(disk db.KeyValueStore, config *Config) (*Database, error) {
 	// Default to path config if not provided
 	if config == nil {
 		config = &Config{
 			PathConfig: &pathdb.Config{},
 		}
 	}
 
+	pathdb, err := pathdb.New(disk, config.PathConfig)
+	if err != nil {
+		return nil, err
+	}
+
 	return &Database{ // TODO: handle both pathdb and hashdb
-		triedb: pathdb.New(disk, config.PathConfig),
+		triedb: pathdb,
 		config: config,
-	}
+	}, nil
 }
 
 func (d *Database) Update(
 	root,
 	parent felt.Felt,
 	blockNum uint64,
-	classNodes map[trieutils.Path]trienode.TrieNode,
-	contractNodes map[felt.Felt]map[trieutils.Path]trienode.TrieNode,
+	mergeClassNodes, mergeContractNodes *trienode.MergeNodeSet,
```

Comment:
This hurts readability in my opinion, please a parameter per line if we are using the column format

Reply:
changed